# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContextFieldList.h
# Commit: a39da23d450b
# Full Hash: a39da23d450b7e3cda053097d77ca47de40f1bb2
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2019-03-15 03:42:43
# Regressor Bug: 1532661
# File Overlap Count: 2
# Description:
#   Bug 1532661 - Part 2: Use a field list macro header for synced fields, r=farre
#   
#   Depends on D22190
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22191
# ==============================================================================

diff -r 84d076d2163d -r a39da23d450b docshell/base/BrowsingContextFieldList.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/docshell/base/BrowsingContextFieldList.h	Thu Mar 14 18:51:05 2019 +0000
@@ -0,0 +1,23 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+MOZ_BC_FIELD(Name, nsString)
+MOZ_BC_FIELD(Closed, bool)
+MOZ_BC_FIELD(CrossOriginPolicy, nsILoadInfo::CrossOriginPolicy)
+
+// The current opener for this BrowsingContext. This field can be skipped by
+// defining `MOZ_BC_FIELD_SKIP_OPENER`, which is used when serializing
+// synchronized field state when initializing browsing contexts.
+#ifndef MOZ_BC_FIELD_SKIP_OPENER
+MOZ_BC_FIELD(Opener, RefPtr<BrowsingContext>)
+#endif
+
+// Toplevel browsing contexts only. This field controls whether the browsing
+// context is currently considered to be activated by a gesture.
+MOZ_BC_FIELD(IsActivatedByUserGesture, bool)
+
+#undef MOZ_BC_FIELD
+#undef MOZ_BC_FIELD_SKIP_OPENER